### DISCUSSION TITLE
Single Supabase project using Google Auth Provider with Web and Flutter Android App

### DISCUSSION THREAD

**Discussion by spetz83** [supabase/supabase#7093] [Questions] [answered]
I currently have a  single supabase project for an app that I have been working on. For the Web frontend, I am using React. So far everything on the react side has been working fine. I was able to configure supabase with my google oauth credentials and I am using google as my auth provider in react. 

I am now attempting to create the same app using Flutter to create an Android (and eventually iOS) version of my web app. I wanted users to be able to fire up a native app while on the go and have access to their data. The issue that I am having is that when I attempt to login using google on the Flutter side of things, I get a 403 from google. My guess would be that this is because the oauth client is meant for web, and not Android, though I might be off base there.

Is it possible to do this? I know I can make a mobile specific oauth client, but it doesn't appear that a single supabase project can have multiple clients associated with it.

Any help would be much appreciated.

**Comment by spetz83** [comment]
Can anyone point me in the right direction on this? I am still stuck. If this isn't possible, I'd love to at least know how others are working around this...

**Comment by GaryAustin1** [comment]
I don't have an answer for you directly.  Have you searched discussions on flutter auth and issues on auth flutter?
In general Supabase auth just cares about email match or phone # match and what "client" is being used should not matter.  If the individual client works with an email, then it should not matter what client was last used by the user.   There is a flutter client https://pub.dev/packages/supabase_flutter that seems to be recommended when I search on "flutter auth" here in discussions.

**Reply by spetz83** [reply]
Thanks for the reply. Maybe I didn't explain correctly. I am using Google Authentication. Their Oauth2 social login. So, I am using the supabase sdk in flutter, trying to use the social auth solution with Google as a provider. Using Supabase on my web app with google as the auth provider works really well. I want to do the same thing with my flutter companion app, bnut I'm not sure if it's possible since it is complaining that I have a "Web" oauth secret, but im trying to use it with a mobile app.

**Reply by GaryAustin1** [reply]
So if you use just Flutter alone your app works with Google as the provider?  But, Google/Supabase Auth requires two different methods that share a common "secret" and Google only allows one method with a common secret?   

Sorry for not being able to help, just trying to move you forward.  Maybe generate an issue in supabase/gotrue github if no one jumps in.

**Comment by dshukertjr** [comment, marked as answer]
Hey @spetz83 !

There is a way of doing  social login using Supabase and Flutter, and that is to redirect user to the browser version of the login page. 

[Here](https://github.com/phamhieu/supabase-flutter-demo) is an example app that signs users in using Github login on Flutter application. The way it works is when the user taps the "Login with Github" button, a browser will open up, and upon sining in, the user is brought back to your Flutter application using deep links. 

We are working on bringing proper mobile auth soon to our SDK in the near future.

**Reply by spetz83** [reply]
Thanks for the reply. I will look into that.
